packages/dev/sharedUiComponents/src/split/splitter.tsx

```diff
@@ -0,0 +1,121 @@
+import { useContext, useEffect, useRef } from "react";
+import styles from "./splitContainer.module.scss";
+import type { ControlledSize } from "./splitContext";
+import { SplitContext, SplitDirection } from "./splitContext";
+
+/**
+ * Splitter component properties
+ */
+export interface ISplitterProps {
+    /**
+     * Unique identifier
+     */
+    id?: string;
+
+    /**
+     * Splitter size
+     */
+    size: number;
+
+    /**
+     * Minimum size for the controlled element
+     */
+    minSize?: number;
+
+    /**
+     * Maximum size for the controlled element
+     */
+    maxSize?: number;
+
+    /**
+     * Initial size for the controlled element
+     */
+    initialSize?: number;
+
+    /**
+     * Defines the controlled side
+     */
+    controlledSide: ControlledSize;
+
+    /**
+     * refObject to the splitter element
+     */
+    refObject?: React.RefObject<HTMLDivElement>;
+}
+
+/**
+ * Creates a splitter component
+ * @param props defines the splitter properties
+ * @returns the splitter component
+ */
+export const Splitter: React.FC<ISplitterProps> = (props) => {
+    const elementRef: React.RefObject<HTMLDivElement> = props.refObject || useRef(null);
+    const splitContext = useContext(SplitContext);
+    let isCaptured = false;
+    let startValue: number;
+
+    useEffect(() => {
+        if (!elementRef.current) {
+            return;
+        }
+
+        if (splitContext.direction === SplitDirection.Horizontal) {
+            elementRef.current.style.width = `${props.size}px`;
+            elementRef.current.style.height = `100%`;
+            elementRef.current.classList.add(styles["horizontal"]);
+        } else {
+            elementRef.current.style.height = `${props.size}px`;
+            elementRef.current.style.width = `100%`;
+            elementRef.current.classList.add(styles["vertical"]);
+        }
+
+        splitContext.init(elementRef.current, props.controlledSide, props.initialSize, props.minSize, props.maxSize);
+    });

```

Comment:
I assume you mean to add `[]` here as a 2nd parameter? or do you want it to run on every re-render?
```suggestion
    }, []);
```

Reply:
I need it on every rerender else I will lose the context on the split container (which stores the various arrays). I tried with [] but then if the control is regenerated the arrays (like isfloating) are empty